src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -859,10 +856,18 @@ def save_as_table(
         child: SnowflakePlan,
         source_plan: Optional[LogicalPlan],
         use_scoped_temp_objects: bool,
-        is_generated: bool,  # true if the table is generated internally
+        creation_source: TableCreationSource,
         child_attributes: List[Attribute],
```

Comment:
seems we relaxed the child attributes here, maybe let's make this filed opitional and pass None when no column definition is needed, and add comment about when this filed will be None.